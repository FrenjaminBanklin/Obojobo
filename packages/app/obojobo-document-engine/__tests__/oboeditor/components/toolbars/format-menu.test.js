import { mount } from 'enzyme'
import React from 'react'

import FormatMenu from '../../../../src/scripts/oboeditor/components/toolbars/format-menu'

const LIST_LINE_NODE = 'ObojoboDraft.Chunks.List.Line'

describe('Format Menu', () => {
	test('FormatMenu node', () => {
		const editor = { current: {
		}}
		const value = {
			selection: { focus: { key: 'mock-key', offset: 1}, anchor: { key: 'mock-key', offset: 1} }
		}

		const component = mount(<FormatMenu editor={editor} value={value}/>)
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('FormatMenu node toggles mark', () => {
		const editor = { current: {
			focus: jest.fn()
		}}
		editor.current.toggleMark = jest.fn().mockReturnValue(editor.current)
		const value = {
			selection: { focus: { key: 'mock-key', offset: 1}, anchor: { key: 'mock-key', offset: 4} }
		}

		const component = mount(<FormatMenu editor={editor} value={value}/>)

		component
			.find('button')
			.at(2)
			.simulate('click')

		expect(editor.current.toggleMark).toHaveBeenCalled()
	})

	test('FormatMenu node calls editor.changeToType for each paraggraph style', () => {
		const editor = { current: {
			changeToType: jest.fn()
		}}
		const value = {
			selection: { focus: { key: 'mock-key', offset: 1}, anchor: { key: 'mock-key', offset: 4} }
		}

		const component = mount(<FormatMenu editor={editor} value={value}/>)

		component
			.find('button')
			.at(12)
			.simulate('click')
		component
			.find('button')
			.at(13)
			.simulate('click')
		component
			.find('button')
			.at(14)
			.simulate('click')
		component
			.find('button')
			.at(15)
			.simulate('click')
		component
			.find('button')
			.at(16)
			.simulate('click')
		component
			.find('button')
			.at(17)
			.simulate('click')
		component
			.find('button')
			.at(18)
			.simulate('click')
		component
			.find('button')
			.at(19)
			.simulate('click')

		expect(editor.current.changeToType).toHaveBeenCalledTimes(8)
	})

	test('FormatMenu node sets indent', () => {
		const editor = { current: {
			value: { blocks: [{ type: LIST_LINE_NODE }]},
			unwrapNodeByKey: jest.fn(),
			unindentList: jest.fn()
		}}
		const value = {
			selection: { focus: { key: 'mock-key', offset: 1}, anchor: { key: 'mock-key', offset: 4} }
		}

		const component = mount(<FormatMenu editor={editor} value={value}/>)

		component
			.find('button')
			.at(25)
			.simulate('click')

		expect(editor.current.unindentList).toHaveBeenCalled()
	})

	test('FormatMenu node calls editor.changeToType for each bullet style', () => {
		const editor = { current: {
			changeToType: jest.fn()
		}}
		const value = {
			selection: { focus: { key: 'mock-key', offset: 1}, anchor: { key: 'mock-key', offset: 4} }
		}

		const component = mount(<FormatMenu editor={editor} value={value}/>)

		// Unordered lists
		component
			.find('button')
			.at(28)
			.simulate('click')
		component
			.find('button')
			.at(29)
			.simulate('click')
		component
			.find('button')
			.at(30)
			.simulate('click')

		// Ordered lists
		component
			.find('button')
			.at(32)
			.simulate('click')
		component
			.find('button')
			.at(33)
			.simulate('click')
		component
			.find('button')
			.at(34)
			.simulate('click')
		component
			.find('button')
			.at(35)
			.simulate('click')
		component
			.find('button')
			.at(36)
			.simulate('click')

		expect(editor.current.changeToType).toHaveBeenCalledTimes(8)
	})
})