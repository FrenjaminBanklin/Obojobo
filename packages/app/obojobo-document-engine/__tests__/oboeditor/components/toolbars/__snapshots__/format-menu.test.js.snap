// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Format Menu FormatMenu node 1`] = `
"<div class=\\"visual-editor--drop-down-menu\\"><div class=\\"dropdown  is-not-open\\" tabindex=\\"-1\\"><button class=\\"menu-title\\">Format</button><div class=\\"menu-items\\"><div class=\\"dropdown  is-not-open\\" tabindex=\\"-1\\"><button class=\\"menu-title\\">Text</button><div class=\\"menu-items\\"><button>Bold<span>
B</span></button><button>Italic<span>
I</span></button><button>Strikethrough<span>
D</span></button><button>Quote<span>
\\"</span></button><button>Monospace<span>
M</span></button><button>Equation<span>
/</span></button><button>Link<span>
K</span></button><button>Superscript<span>
.</span></button><button>Subscript<span>
,</span></button></div></div><div class=\\"dropdown  is-not-open\\" tabindex=\\"-1\\"><button class=\\"menu-title\\">Paragraph styles</button><div class=\\"menu-items\\"><button>Normal Text<span>
Ctrl+Shift+Space</span></button><button>Heading 1<span>
Ctrl+Shift+1</span></button><button>Heading 2<span>
Ctrl+Shift+2</span></button><button>Heading 3<span>
Ctrl+Shift+3</span></button><button>Heading 4<span>
Ctrl+Shift+4</span></button><button>Heading 5<span>
Ctrl+Shift+5</span></button><button>Heading 6<span>
Ctrl+Shift+6</span></button><button>Code<span>
Ctrl+Shift+C</span></button></div></div><div class=\\"dropdown  is-not-open\\" tabindex=\\"-1\\"><button class=\\"menu-title\\">Align &amp; indent</button><div class=\\"menu-items\\"><button>Left Align<span>
L</span></button><button>Center Align<span>
R</span></button><button>Right Align<span>
E</span></button><button>Indent<span>
Alt+Tab</span></button><button>Unindent<span>
Shift+Tab</span></button><button>Hanging Indent<span>
H</span></button></div></div><div class=\\"dropdown  is-not-open\\" tabindex=\\"-1\\"><button class=\\"menu-title\\">Bullets &amp; numbering</button><div class=\\"menu-items\\"><div class=\\"dropdown  is-not-open\\" tabindex=\\"-1\\"><button class=\\"menu-title\\">Bulleted List</button><div class=\\"menu-items\\"><button>● Disc<span>
Ctrl+Shift+K</span></button><button>○ Circle</button><button>■ Square</button></div></div><div class=\\"dropdown  is-not-open\\" tabindex=\\"-1\\"><button class=\\"menu-title\\">Numbered List</button><div class=\\"menu-items\\"><button>Numbers<span>
Ctrl+Shift+L</span></button><button>Lowercase Alphabet</button><button>Lowercase Roman Numerals</button><button>Uppercase Alphabet</button><button>Uppercase Roman Numerals</button></div></div></div></div></div></div></div>"
`;
